controllers/model/model_controller.go

```diff
@@ -210,12 +211,14 @@ func (r *ModelReconciler) initializeContext(ctx *reconcileRequestContext) error
 	ctx.SageMakerClient = clientwrapper.NewSageMakerClientWrapper(r.createSageMakerClient(awsConfig))
 	ctx.Log.Info("Loaded AWS config")
 
-	if ctx.Model.Spec.PrimaryContainer.Mode == nil {
-		ctx.Model.Spec.PrimaryContainer.Mode = aws.String(DefaultContainerDefinitionMode)
+	if ctx.Model.Spec.PrimaryContainer != nil {
+		if ctx.Model.Spec.PrimaryContainer.Mode == nil {
```

Comment:
Can we combine this to 	if ctx.Model.Spec.PrimaryContainer != nil &&
		 ctx.Model.Spec.PrimaryContainer.Mode == nil 

?